witchcraft/loggers.go

```diff
@@ -141,17 +162,43 @@ func (s *Server) initWrappedLoggers(useConsoleLog bool, productName, productVers
 // Returns a io.Writer that can be used as the underlying writer for a logger.
 // If either logToStdout or logToStdoutBasedOnEnv() is true, then stdoutWriter is returned.
 // Otherwise, a default writer that writes to slsFilename is returned.
-func newDefaultLogOutputWriter(slsFilename string, logToStdout bool, stdoutWriter io.Writer) io.Writer {
+func newDefaultLogOutputWriter(slsFilename string, logToStdout bool, stdoutWriter io.Writer) (io.Writer, func(ctx context.Context)) {
 	if logToStdout || logToStdoutBasedOnEnv() {
-		return stdoutWriter
+		return stdoutWriter, nil
 	}
-	return &lumberjack.Logger{
+	logger := &lumberjack.Logger{
 		Filename:   fmt.Sprintf(defaultLogOutputFormat, slsFilename),
 		MaxSize:    1000,
 		MaxBackups: 10,
 		MaxAge:     30,
 		Compress:   true,
 	}
+	flusher := func(ctx context.Context) {
+		periodicallyRotateLogFiles(ctx, logger, slsFilename)
+	}
+	return logger, flusher
+}
+
+func periodicallyRotateLogFiles(ctx context.Context, logger *lumberjack.Logger, slsFilename string) {
```

Comment:
nit: `periodicallyRotateLogFile` since it is only doing one file

Reply:
Fixed.